va/va.go

```diff
@@ -452,15 +454,35 @@ func (va *ValidationAuthorityImpl) observeLatency(op, perspective, challType, pr
 	va.metrics.validationLatency.With(labels).Observe(latency.Seconds())
 }
 
-// performRemoteValidation coordinates the whole process of kicking off and
-// collecting results from calls to remote VAs' PerformValidation function. It
-// returns a problem if too many remote perspectives failed to corroborate
-// domain control, or nil if enough succeeded to surpass our corroboration
-// threshold.
-func (va *ValidationAuthorityImpl) performRemoteValidation(
-	ctx context.Context,
-	req *vapb.PerformValidationRequest,
-) *probs.ProblemDetails {
+// remoteOperation is a func type that encapsulates the operation and request
+// passed to va.performRemoteOperation. The operation must be a method on
+// vapb.VAClient or vapb.CAAClient, and the request must be the corresponding
+// proto.Message passed to that method.
+type remoteOperation = func(context.Context, RemoteVA, proto.Message) (remoteResult, error)
+
+// remoteResult is an interface that must be implemented by the results of a
+// remoteOperation, such as *vapb.ValidationResult and *vapb.IsCAAValidResponse.
+// It provides methods to access problem details, the associated perspective,
+// and the RIR.
+type remoteResult interface {
+	proto.Message
+	GetProblem() *corepb.ProblemDetails
+	GetPerspective() string
+	GetRir() string
+}
+
+var _ remoteResult = (*vapb.ValidationResult)(nil)
+var _ remoteResult = (*vapb.IsCAAValidResponse)(nil)
+
+// performRemoteOperation performs the provided operation and request on all
+// remote VAs. It cancels the operation and returns early if either the required
+// number of successful results is obtained or the number of failures exceeds
+// va.maxRemoteFailures.
```

Comment:
Proposed comment tweak:

```suggestion
// performRemoteOperation concurrently calls the provided operation with `req` and a
// RemoteVA once for each configured RemoteVA. It cancels remaining operations and returns
// early if either the required number of successful results is obtained or the number of
// failures exceeds va.maxRemoteFailures.
```

Explanation: strictly speaking, `performRemoteOperation` isn't responsible for calling a method on the RemoteVA - that's done by the callback.

Also, it's more precise to say "cancels remaining operations" than "cancels the operation", because we're specifically talking about the case where some operations have succeeded or failed and some remain pending.

Reply:
Ah, I see what you're saying. Thanks for providing an improved comment!